addons/html_builder/static/src/core/building_blocks/builder_number_input.js

```diff
@@ -79,13 +79,20 @@ export class BuilderNumberInput extends Component {
                 /(?<savedValue>[\d.e+-]+)(?<savedUnit>\w*)/
             ).groups;
             if (savedUnit || this.props.saveUnit) {
+                const numericSavedValue = parseFloat(savedValue);
+                if (Number.isNaN(numericSavedValue)) {
+                    return value;
+                }
```

Comment:
Why do you need to display something that is not a `number`in `BuilderNumberInput`?